gateway/proxy_muxer.go

```diff
@@ -45,6 +45,9 @@ func (h *h2cWrapper) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 func (h *handleWrapper) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// make request body to be nopCloser and re-readable before serve it through chain of middlewares
 	nopCloseRequestBody(r)
+	defer func() {
+		r.Body = nil
+	}()
```

Comment:
I suspect this isn't the fix, just an attempt to trigger the GC. Needs retest on removal.

Reply:
Pete confirmed with re-test, this needs removal.

Reply:
To be confirmed - there are still grpc OOM cases, coordinating with Pete.